omnigibson/prims/rigid_prim.py

```diff
@@ -320,16 +326,29 @@ def set_position_orientation(self, position=None, orientation=None):
         self._rigid_prim_view.set_world_poses(positions=position, orientations=orientation)
         PoseAPI.invalidate()
 
-    def get_position_orientation(self):
+    def get_position_orientation(self, clone=True):
+        """
+        Gets prim's pose with respect to the world's frame.
+
+        Args:
+            clone (bool): Whether to clone the internal buffer or not when grabbing data
+
+        Returns:
+            2-tuple:
+                - 3-array: (x,y,z) position in the world frame
+                - 4-array: (x,y,z,w) quaternion orientation in the world frame
+        """
         # Return cached pose if we're kinematic-only
         if self.kinematic_only and self._kinematic_world_pose_cache is not None:
             return self._kinematic_world_pose_cache
 
-        pos, ori = self._rigid_prim_view.get_world_poses()
+        pos, ori = self._rigid_prim_view.get_world_poses(clone=clone)
 
-        assert np.isclose(
-            np.linalg.norm(ori), 1, atol=1e-3
-        ), f"{self.prim_path} orientation {ori} is not a unit quaternion."
+        if gm.DEBUG:
+            # Only run this assert when debugging since this actually results in some nontrivial slowdown
+            assert np.isclose(
+                np.linalg.norm(ori), 1, atol=1e-3
+            ), f"{self.prim_path} orientation {ori} is not a unit quaternion."
```

Comment:
I'm guessing it's actually the sqrt that's slow in the norm computation. Can we replace it with `np.sum(ori * ori)` instead? Should still equal 1 although the atol might need to be decreased. I'd really like to keep this assertion because it catches physx crashes right after they happen.

Reply:
Just timed it. Changing to squared norm doesn't actually impact it that much. Looks like the most expensive part is actually the `np.isclose()` check.

Reply:
Do we have an idea about when the quat is invalid? i.e: are all values nans, or all 0s, or something? maybe that's a faster check

Reply:
e.g: `any(quat)` --> this is 2 orders of magnitude faster than `np.isclose()`

Reply:
Can we just change this to np.sum(ori  * ori) - 1 < 1e-3 ?

Reply:
yep, this works!

Reply:
actually this doesn't work, need to also check for negative side too. added that as well.